src/runtime/TemporalObject.h

```diff
@@ -351,6 +352,128 @@ class TemporalZonedDateTime : public Temporal {
     TemporalCalendar* m_calendar;
 };
 
+class TemporalDuration : public Temporal {
+public:
+    explicit TemporalDuration(ExecutionState& state, int years, int months, int weeks, int days, int hours, int minutes, int seconds, int milliseconds, int microseconds, int nanoseconds);
+    explicit TemporalDuration(ExecutionState& state, Object* proto, int years, int months, int weeks, int days, int hours, int minutes, int seconds, int milliseconds, int microseconds, int nanosecond);
+    static bool isValidDuration(const std::array<int, 10>& fields);
+    static int durationSign(const std::array<int, 10>& fields);
+    static std::map<std::string, int> createDurationRecord(ExecutionState& state, int years, int months, int weeks, int days, int hours, int minutes, int seconds, int milliseconds, int microseconds, int nanoseconds);
+    static Value createTemporalDuration(ExecutionState& state, int years, int months, int weeks, int days, int hours, int minutes, int seconds, int milliseconds, int microseconds, int nanoseconds, Optional<Object*> newTarget);
+
+    static Value toTemporalDuration(ExecutionState& state, const Value& item);
+    static std::map<std::string, int> toTemporalDurationRecord(ExecutionState& state, const Value& temporalDurationLike);
+    static std::map<std::string, Value> toTemporalPartialDurationRecord(ExecutionState& state, const Value& temporalDurationLike);
+    static Value createNegatedTemporalDuration(ExecutionState& state, const Value& duration);
+
+    bool isTemporalDurationObject() const override
+    {
+        return true;
+    }
+
+    int getYear() const
```

Comment:
No newline after the getters from here. Add everywhere, or remove the newline above.